both/methods.js

```diff
@@ -153,16 +153,25 @@ Meteor.methods({
 
     Meteor.call('refreshRoundCodes', competitionId, eventCode);
   },
-  addNonEventRound: function(competitionId, round) {
+  addScheduleEvent: function(competitionId, eventData, roundId) {
     check(competitionId, String);
     throwIfCannotManageCompetition(this.userId, competitionId);
-    return Rounds.insert({
+
+    var round = Rounds.findOne(roundId);
+
+    return ScheduleEvents.insert({
       competitionId: competitionId,
-      title: round.title,
-      startMinutes: round.startMinutes,
-      durationMinutes: round.durationMinutes,
+      roundId: (round ? round._id : null),
```

Comment:
Ah. I see. These were not written with the intention that they propagate out beyond the callback function. As you metnioned, they are executed asynchronously, so any escaping exceptions could not be handled in the function that calls `Meteor.call` anyways. As https://github.com/meteor/meteor/issues/1640 says, meteor would have to provide some hook for us to give some code to be run on uncaught exceptions.

Anyways, that's not the point of these throws. It would be better to think of them as production asserts. They should never fire, but if they do, IMO, it's better than silently doing nothing, as the code you suggested would do:

```javascript
if(!err) {
  $tr.removeClass('saving');
}
```

My understanding is that when we throw an error, we at least get something printed to the console, which I would hope would aid in debugging.